Sources/Subs-Db-postgresql.php

```diff
@@ -776,22 +788,47 @@ function smf_db_insert($method = 'replace', $table, $columns, $data, $keys, $dis
 		foreach ($data as $dataRow)
 			$insertRows[] = smf_db_quote($insertData, array_combine($indexed_columns, $dataRow), $connection);
 
-		foreach ($insertRows as $entry)
-			// Do the insert.
-			$smcFunc['db_query']('', '
-				INSERT INTO ' . $table . '("' . implode('", "', $indexed_columns) . '")
-				VALUES
-					' . $entry.$replace,
-				array(
-					'security_override' => true,
-					'db_error_skip' => $method == 'ignore' || $table === $db_prefix . 'log_errors',
-				),
-				$connection
-			);
+		// Do the insert.
+		$request = $smcFunc['db_query']('', '
+			INSERT INTO ' . $table . '("' . implode('", "', $indexed_columns) . '")
+			VALUES
+				' . implode(',
+				', $insertRows).$replace.$returning,
+			array(
+				'security_override' => true,
+				'db_error_skip' => $method == 'ignore' || $table === $db_prefix . 'log_errors',
+			),
+			$connection
+		);
+
+		if ($with_returning && $request !== false)
+		{
+			if ($returnmode === 2)
+				$return_var = array();
+
+			while($row = $smcFunc['db_fetch_assoc']($request) && $with_returning)
+			{
+				if (is_int($row[$keys[0]])) // try to emulate mysql limitation
```

Comment:
I'm not 100% sure if I should do this,
but this in my eyes the only way to check if the fields is a ai field(int).

The main fokus is for person how develop something on smf with pg and
hop the stuff is running on mysql too.
But on the other side maybe is this behavior for this mod/addon (not only we develop for smf)
not important...